platform/kube/queue.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/pkg/util/flowcontrol"
+)
+
+// Queue of work tickets processed using a rate-limiting loop
+type Queue interface {
+	// Push a ticket
+	Push(Task)
+	// Run the loop until a signal on the channel
+	Run(chan struct{})
+}
+
+// Task object for the event watchers; processes until handler succeeds
+type Task struct {
```

Comment:
The `obj` argument is coming from the event watcher and needs to be passed to `func`.
I'll probably need to add another argument for the event type (`add`, `remove`, `update`).